### PR TITLE
update `ecs` and add `cloudmap-namespace` component

### PR DESCRIPTION
**Pull request by Benbentwo**
## what

- New Cloudmap namespace
- integration with ECS

### CODE REVIEW

File: modules/cloudmap-namespace/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  value = module.this.id
+}
+
+output "id" {
+  value = coalesce(one(aws_service_discovery_http_namespace.default[*].id), one(aws_service_discovery_private_dns_namespace.default[*].id), one(aws_service_discovery_public_dns_namespace.default[*].id))
+}
+
+output "arn" {
+  value = coalesce(one(aws_service_discovery_http_namespace.default[*].arn), one(aws_service_discovery_private_dns_namespace.default[*].arn), one(aws_service_discovery_public_dns_namespace.default[*].arn))
+}
```
Comment (milldr) [inline modules/cloudmap-namespace/outputs.tf:11]: please add descriptions to each output. tflint requirement

File: modules/ecs-service/cloud-map.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#  value = local.service_config_with_id
+#}
+#
+#output "service_discovery" {
+#  value = local.service_discovery
+#}
+#
+#output "aws_service_discovery_service" {
+#  value = aws_service_discovery_service.default
+#}
```
Comment (milldr) [inline modules/ecs-service/cloud-map.tf:71]:
```suggestion
```

File: modules/ecs-service/cloud-map.tf [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  version = "1.5.0"
+
+  component = each.key
+
+  # we ignore errors because the namespace may be a name or an arn of a namespace for the service.
+  ignore_errors = true
+  context       = module.this.context
+}
+
+locals {
```
Comment (milldr) [inline modules/ecs-service/cloud-map.tf:14]: can we merge locals here and on line 32?

Comment (Benbentwo) [reply modules/ecs-service/cloud-map.tf:16]: I'd prefer not to, they are for two different methods of connecting ecs services, either Service Discovery or Service Connect.

**Review by milldr — APPROVED** [review]

**Pull request merged**